csclient/csclient.go

```diff
@@ -174,37 +177,42 @@ func (c *Client) GetArchive(id *charm.URL) (r io.ReadCloser, eid *charm.URL, has
 	}
 	resp, err := c.Do(req, u.String())
 	if err != nil {
-		return nil, nil, "", 0, errgo.NoteMask(err, "cannot get archive", isAPIError)
+		terr := params.MaybeTermsAgreementError(err)
+		if err1, ok := errgo.Cause(terr).(*params.TermAgreementRequiredError); ok {
+			terms := strings.Join(err1.Terms, " ")
+			return fail(errgo.Newf(`cannot get archive: please agree to the following terms %s. Try: "juju agree %s"`, terms, terms))
```

Comment:
Rather than repeating the terms in the error message, how about:

   `cannot get archive because some terms have not been agreed to. Try "juju agree %s"`

?